src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -115,8 +116,22 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
 
     if deployment_source_url and deployment_local_git:
         raise MutuallyExclusiveArgumentError('usage error: --deployment-source-url <url> | --deployment-local-git')
+    if deployment_container_image_name and container_image_name:
+        raise MutuallyExclusiveArgumentError('usage error: --deployment-container-image-name | --container-image-name')
+    if container_registry_url and not container_image_name:
+        raise ArgumentUsageError("Please use --container-image-name to provide the image name")
 
-    docker_registry_server_url = parse_docker_image_name(deployment_container_image_name)
+    if container_registry_url:
+        container_registry_url = parse_container_registry_url(container_registry_url)
+    else:
+        container_registry_url = parse_docker_image_name(deployment_container_image_name)
```

Comment:
rename the methos to parse_container_image_name as we are trying to get rid of "docker" when possible.

Reply:
Will do this renaming in another PR